app.py

```diff
@@ -0,0 +1,437 @@
+            update_parent_reservation("prod", knack_record_id, today_date)
+        knack_response = requests.put(
+            f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID}/records/{knack_record_id}",
+            headers=headers["prod"],
+            data=knack_payload,
+        )
+    if knack_response.status_code == 200:
+        return "Payment status updated", knack_response.status_code
+    # if unsuccessful, return knack's status response as response
+    return knack_response.text, knack_response.status_code
+
+
+@app.route("/citybase_postback_uat", methods=["POST"])
+def handle_postback_uat():
+    today_date = datetime.now().strftime("%m/%d/%Y %H:%M")
+    citybase_data = request.get_json()
+
+    knack_record_id = get_knack_record_id(citybase_data)
+    # if knack_record_id is not a string, then it is an error tuple
+    if not isinstance(knack_record_id, str):
+        return knack_record_id
+
+    knack_invoice = get_knack_invoice(citybase_data)
+    # if knack_record_id is not a string, then it is an error tuple
+    if not isinstance(knack_invoice, str):
+        return knack_invoice
+
+    payment_status = citybase_data["data"]["status"]
+    payment_amount = citybase_data["data"]["total_amount"]
+    citybase_id = citybase_data["data"]["id"]
+
+    message_payload = create_message_json(
+        "uat", citybase_id, today_date, knack_invoice, payment_status
+    )
+    requests.post(
+        f"{KNACK_API_URL}{MESSAGES_OBJECT_ID_UAT}/records/",
+        headers=headers["uat"],
+        data=message_payload,
+    )
+
+    # if a refund, post a new record to knack transactions table
+    if payment_status == "refunded":
+        knack_payload = get_knack_refund_payload(
+            "uat",
+            payment_status,
+            payment_amount,
+            knack_invoice,
+            today_date,
+            knack_record_id,
+        )
+        knack_response = requests.post(
+            f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID_UAT}/records/",
+            headers=headers["uat"],
+            data=knack_payload,
+        )
+    # otherwise, update existing record payment status
+    else:
+        knack_payload = get_knack_payload("uat", payment_status, today_date)
+        if payment_status == "successful":
+            # if this was a successful payment we need to update reservation record
+            update_parent_reservation("uat", knack_record_id, today_date)
+        knack_response = requests.put(
+            f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID_UAT}/records/{knack_record_id}",
+            headers=headers["uat"],
+            data=knack_payload,
+        )
+    if knack_response.status_code == 200:
+        return "Payment status updated", knack_response.status_code
+    # if unsuccessful, return knack's status response as response
+    return knack_response.text, knack_response.status_code
+
+
+if __name__ == "__main__":
+    # todo: remember to turn off debug!
+    app.run(debug=True, host="0.0.0.0")
```

Comment:
One thing I think we might consider as part of dockerizing the application with a production-ready web server would be to run it in ECS Fargate. We'd get a reduction in monthly cost because we could get rid of the EC2 instance it's on and also pick up scalability, even though that's not really a near-term need on this app. I don't think we'll need a reverse proxy on this one as John notes with it being behind an elastic load balancer, but nginx or haproxy could be used if we did.